fuse_constraints/include/fuse_constraints/normal_delta_pose_2d_cost_functor.h

```diff
@@ -112,14 +117,18 @@ bool NormalDeltaPose2DCostFunctor::operator()(
 {
   Eigen::Map<const Eigen::Matrix<T, 2, 1>> position1_vector(position1);
   Eigen::Map<const Eigen::Matrix<T, 2, 1>> position2_vector(position2);
-  Eigen::Map<Eigen::Matrix<T, 3, 1>> residuals_matrix(residual);
-  residuals_matrix.template head<2>() =
+  Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, 1>> residuals_matrix(residual, A_.rows());
```

Comment:
I would move this line to right before `A_.template cast<T>() * full_residuals_matrix;`
That is the only place it is used; no reason to clutter up my mind with the extra variable name until it is needed.